cwltool/provenance.py

```diff
@@ -448,41 +448,34 @@ def evaluate(self,
                  process,           # type: Process
                  job,               # type: Any
                  job_order_object,  # type: Dict[Text, Text]
-                 make_fs_access,    # type: Callable[[Text], StdFsAccess]
                  research_obj       # type: ResearchObject
-                ):  # type: (...) -> Optional[str]
-        """Evaluate the nature of job and initialize the activity start."""
-        process_run_id = None
-        research_obj.make_fs_access = make_fs_access
+                ):  # type: (...) -> None
+        """Evaluate the nature of job"""
         if not hasattr(process, "steps"):
-            # record provenance of an independent commandline tool execution
+            # record provenance of independent commandline tool executions
             self.prospective_prov(job)
             customised_job = copy_job_order(job, job_order_object)
             self.used_artefacts(customised_job, self.workflow_run_uri)
             research_obj.create_job(customised_job, job)
-            # self.used_artefacts(inputs, self.workflow_run_uri)
-            name = ""
-            if hasattr(job, "name"):
-                name = str(job.name)
-            process_name = urllib.parse.quote(name, safe=":/,#")
-            process_run_id = self.workflow_run_uri
         elif hasattr(job, "workflow"):
-            # record provenance for the workflow execution
+            # record provenance of workflow executions
             self.prospective_prov(job)
             customised_job = copy_job_order(job, job_order_object)
             self.used_artefacts(customised_job, self.workflow_run_uri)
-            # inputs = research_obj.create_job(customised_job)
-            # self.used_artefacts(inputs, self.workflow_run_uri)
-        else:  # in case of commandline tool execution as part of workflow
-            name = ""
-            if hasattr(job, "name"):
-                name = str(job.name)
+
+    def record_process_start(self, process, job, process_run_id=None):
+        # type: (Process, Any, str) -> Optional[str]
+        if not hasattr(process, 'steps'):
+            process_run_id = self.workflow_run_uri
+        elif not hasattr(job, 'workflow'):
+            # commandline tool execution as part of workflow
+            name = str(job.name) if hasattr(job, 'name') else ''
             process_name = urllib.parse.quote(name, safe=":/,#")
-            process_run_id = self.start_process(process_name)
+            process_run_id = self.start_process(process_name, datetime.datetime.now())
         return process_run_id
 
-    def start_process(self, process_name, process_run_id=None):
-        # type: (Any, str, str) -> str
```

Comment:
Odd change from `str` to `Text` - but this type declaration was wrong to start with and should never have had `Any` - probably it has to be `Text` as a `process_name` ultimately will have come from a YAML parsing of a .cwl file, which can have Unicode `id` values.